lib/Conversion/TorchToLinalg/TorchToLinalg.cpp

```diff
@@ -3451,6 +3451,65 @@ class ConvertAtenContiguousOp : public OpConversionPattern<AtenContiguousOp> {
 };
 } // namespace
 
+namespace {
+struct ConvertAtenScalarToTensorLike : ConversionPattern {
+  ConvertAtenScalarToTensorLike(TypeConverter &typeConverter,
+                                MLIRContext *context)
+      : ConversionPattern(typeConverter, MatchAnyOpTypeTag(), /*benefit=*/1,
+                          context) {}
+
+  LogicalResult
+  matchAndRewrite(Operation *op, ArrayRef<Value> operands,
+                  ConversionPatternRewriter &rewriter) const override {
+    if (!isa<AtenTensorIntOp, AtenTensorFloatOp>(op))
+      return rewriter.notifyMatchFailure(
+          op, "not a supported Scalar to Tensor like op");
+
+    if (failed(verifyLinalgCompatibleTypes(op, rewriter)))
+      return failure();
+    Location loc = op->getLoc();
+
+    Value elemVal, dtype, device, requires_grad;
+
+    if (AtenTensorIntOp tensorIntOp = dyn_cast<AtenTensorIntOp>(op)) {
+      AtenTensorIntOp::Adaptor adaptor(operands);
+      elemVal = adaptor.t();
+      dtype = tensorIntOp.dtype();
+      device = tensorIntOp.device();
+      requires_grad = tensorIntOp.requires_grad();
+    }
+    if (AtenTensorFloatOp tensorFloatOp = dyn_cast<AtenTensorFloatOp>(op)) {
+      AtenTensorFloatOp::Adaptor adaptor(operands);
+      elemVal = adaptor.t();
+      dtype = tensorFloatOp.dtype();
+      device = tensorFloatOp.device();
+      requires_grad = tensorFloatOp.requires_grad();
```

Comment:
Since all the members are same in both the operations, you should templatize it.

Reply:
This will evolve in the future and may contain ops like `AtenTensorOp` where some of the members' names change. Hence fingers crossed.